### ISSUE TITLE
Average price in ecommerce product view's total row shows sum instead of average

### ISSUE THREAD

**Issue by RaphaelBergmann** [matomo-org/matomo#16441]
This should be the _average_ average price, not the sum of all average prices.
![image](https://user-images.githubusercontent.com/2065979/93219464-b34dc080-f76b-11ea-9160-bda52a82b8ca.png)

**Comment by tsteur** [comment]
Thanks for reporting this @RaphaelBergmann I tried to reproduce this in various of our sites but couldn't. 

It also seems to work in our public demo https://demo.matomo.org/index.php?module=CoreHome&action=index&idSite=62&period=day&date=yesterday#?idSite=62&period=day&date=yesterday&segment=&category=Goals_Ecommerce&subcategory=Goals_Products

Can you maybe sort by average price and check if there are some high priced items in one of the 622 entries?

**Comment by RaphaelBergmann** [comment]
I sorted the table, but there're no high priced items... We also have [this issue here with sorting](https://github.com/matomo-org/matomo/issues/8556), which could refer to values being stored incorrectly in the DB (sth about number formatting, 12.345,67 €?). We have a very basic installation with just two plugins (FormAnalytics and MarketingCampaignsReporting) but disabling both of them doesn't fix our two issues. Or maybe it has something to do with MariaDB handling things slightly differently?
![image](https://user-images.githubusercontent.com/2065979/93300509-36682880-f7f7-11ea-99e0-0b5823b5dc8e.png)

**Comment by tsteur** [comment]
@RaphaelBergmann can you reproduce the issue on our demo? https://demo.matomo.org/index.php?module=CoreHome&action=index&idSite=62&period=day&date=yesterday#?idSite=62&period=day&date=yesterday&segment=&category=Goals_Ecommerce&subcategory=Goals_Products

If the sorting is an issue to not show the high prices maybe try to select in the bottom right "500 rows" and then you should be able to see all prices over two pages so it might be easier to find any high prices (if there are any).

**Comment by RaphaelBergmann** [comment]
Your demo is running fine, I can't reproduce my issue there.
I checked for high prices again, but couldn't find any. Even checked the table _matomo_log_conversion_item_.
I thought about things that are specific to our shop: We have items that cost 0, where its average price would be 0 too, and perhaps its a 0 / NULL problem. Perhaps you could add a $0 item to your demo shop?
Also, we have 0 as well as negative revenues in _matomo_log_conversion_, e.g. if someone uses a voucher.

**Comment by tsteur** [comment]
@RaphaelBergmann how are you tracking the ecommerce data? 

Are you viewing the "Ecommerce orders" view there? 
![image](https://user-images.githubusercontent.com/273120/93537071-a49b1f80-f99e-11ea-935a-d64badf940fa.png)

Is there any segment applied? I checked... generally the price is divided by the number of orders and I'm seeing a lot of products there but barely any orders so I'm assuming the problem is maybe that the average should only sum prices of products with orders vs in your case it might also sum products without orders. Might be related to tracking "product views".

**Comment by RaphaelBergmann** [comment]
We use JavaScript for eCommerce tracking, e.g. this for product views:
![image](https://user-images.githubusercontent.com/2065979/93587946-f5187a00-f9aa-11ea-84d3-e5ccc542bbdf.png)
and this for conversion tracking (on the "thankyou"-page):
![image](https://user-images.githubusercontent.com/2065979/93587490-483dfd00-f9aa-11ea-9e21-be2df1ca9989.png)
My screenshots above all show the "Ecommerce Orders", not the "Abandoned Carts". I was already wondering about items with 0 purchases in the products list. However, the "Abandoned Carts" view shows the same behavior:
![image](https://user-images.githubusercontent.com/2065979/93588444-b636f400-f9ab-11ea-9cdd-cf132c998b36.png)
Applying a segment de- or increases the numbers as expected, but even without a segment, the average item prices are still summed.

**Comment by tsteur** [comment]
@RaphaelBergmann is there any chance we could get view access to this site to try and understand things better? It's still not quite clear to me how this happens. The tracking requests itself look good to me and shouldn't cause an issue. If that's possible, could you create a log in for us for hello at matomo.org and send the details to that email address?

**Comment by RaphaelBergmann** [comment]
@tsteur yes, there is a chance - I just sent an email regarding this... Thanks so far :)

**Comment by RaphaelBergmann** [comment]
@tsteur I just sent you the credentials :)

**Comment by tsteur** [comment]
Thanks @RaphaelBergmann it appears to me that it is related to tracking product views. Like Matomo would some up all the product prices then device it by the number of orders. And it would also sum the price of products that had no order for example.
